File: 프로그래머스/3/12907. 거스름돈/거스름돈.py
def solution(n, money):
    method = [1] + [0] * n
    
    for coin in money:
        for i in range(coin, n + 1):
            method[i] = (method[i] + method[i - coin]) % 1_000_000_007
    
    return method[n]

File: 프로그래머스/3/12907. 거스름돈/거스름돈.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> money) {
    int mod = 1'000'000'007;
    vector<int> dp(n + 1, 0);
    dp[0] = 1;
    for(int coin : money){
        for(int i = 0; i + coin <= n; i++){
            dp[i + coin] = (dp[i + coin] + dp[i]) % mod;
        }
    }

    return dp[n];
}